File: advent-of-code/advent_of_code_2016/day_5/part_1.js
const md5 = require('md5');

var input = 'abc';
var index = 0;
var password = '';
var validHashPattern = /^00000.+/;
var hash;

while (password.length < 8) {
  hash = md5(input + index);
  if (validHashPattern.test(hash)) password += hash[5];
  index += 1;
}

console.log(password);

File: advent-of-code/advent_of_code_2016/day_5/part_1.rb
require 'digest'

input = 'reyedfim'
index = 0
password = ''

loop do
  hash = Digest::MD5.hexdigest(input + index.to_s)
  password << hash[5] if hash.match?(/\A00000.+/)
  break if password.length == 8
  index += 1
end

p password
